### PR TITLE
Catch interpreter panics in REPL

### LINKED ISSUE #2027: Catch panics in REPL
### Issue To Be Solved

Part of #2023.

Currently, panics raised by the interpreter aren't caught in the REPL environment. This is inconvenient, since the main usecase for the REPL is rapidly testing expressions and new code. 

### Suggested Solution

Catch panics raised by the interpreter, printing any errors that arose. External panics such as OS signals, go runtime panics, etc. should be re-thrown.

### PR DESCRIPTION
**Pull request by dreamsmasher**
Closes #2027

## Description

Catches interpreter-thrown panics in the REPL, fixing the long-standing issue of out of bounds accesses or variable declarations completely crashing the program.
______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

**Comment by turbolent** [comment]
@dreamsmasher TODO list says issue has label, but issue has no label

File: runtime/pretty/print.go [hunk 1]
```diff
@@ -315,6 +315,10 @@ func (p ErrorPrettyPrinter) writeCodeExcerpts(
 				indicatorLength = maxLineLength
 			}
 
+			if indicatorLength >= len(line) {
+				indicatorLength = len(line)
+			}
```
Comment (turbolent) [inline runtime/pretty/print.go:320]: Likely related to #2021. The pretty printer should not gracefully handle invalid input, we should fix the underlying problem resulting in invalid input

```suggestion
```

**Review by turbolent — COMMENTED** [review]
Nice!

**Review by SupunS — APPROVED** [review]

**Review by SupunS — COMMENTED** [review]
Linter is failing

**Comment by dreamsmasher** [comment]
> Looks good, but please see [#2028 (comment)](https://github.com/onflow/cadence/pull/2028#issuecomment-1268879256)

double whoops

**Pull request merged**